src/MainWindow.vala

```diff
@@ -903,13 +906,22 @@ namespace Terminal {
 
         private void action_open_in_browser () requires (current_terminal != null) {
             var uri = get_current_selection_link_or_pwd ();
-            string? to_open = Utils.sanitize_path (uri, current_terminal.get_shell_location ());
-            if (to_open != null) {
-                try {
-                    Gtk.show_uri_on_window (null, to_open, Gtk.get_current_event_time ());
-                } catch (GLib.Error error) {
-                    warning ("Could not show %s - %s", to_open, error.message);
-                }
+            var to_open = Utils.sanitize_path (uri, current_terminal.get_shell_location (), true);
+            AppInfo? appinfo = get_default_app_for_uri (to_open);
```

Comment:
AppInfo has a static method for launching from the default. We don't need to get the default separately ahead of time https://valadoc.org/gio-2.0/GLib.AppInfo.launch_default_for_uri_async.html